GVFS/GVFS.Common/Prefetch/Git/PrefetchLibGit2Repo.cs

```diff
@@ -42,38 +36,9 @@ public virtual bool TryCopyBlobToFile(string sha, IEnumerable<string> destinatio
                             byte* originalData = Native.Blob.GetRawContent(objHandle);
                             long originalSize = Native.Blob.GetRawSize(objHandle);
 
-                            foreach (string destination in destinations)
-                            {
-                                try
-                                {
-                                    using (SafeFileHandle fileHandle = OpenForWrite(this.Tracer, destination))
-                                    {
-                                        if (fileHandle.IsInvalid)
-                                        {
-                                            throw new Win32Exception(Marshal.GetLastWin32Error());
-                                        }
-
-                                        byte* data = originalData;
-                                        long size = originalSize;
-                                        uint written = 0;
-                                        while (size > 0)
-                                        {
-                                            uint toWrite = size < uint.MaxValue ? (uint)size : uint.MaxValue;
-                                            if (!Native.WriteFile(fileHandle, data, toWrite, out written, IntPtr.Zero))
-                                            {
-                                                throw new Win32Exception(Marshal.GetLastWin32Error());
-                                            }
-
-                                            size -= written;
-                                            data = data + written;
-                                        }
-                                    }
-                                }
-                                catch (Exception e)
-                                {
-                                    this.Tracer.RelatedError("Exception writing {0}: {1}", destination, e);
-                                    throw;
-                                }
+                            foreach (PathWithMode destination in destinations)
+                            {
+                                GVFSPlatform.Instance.FileSystem.WriteFile(this.Tracer, originalData, originalSize, destination.Path, destination.Mode);
```

Comment:
How ugly would it be to leave all of the `unsafe` code in this class (for both platforms)?

That would avoid having to spread `unsafe` to so many places.

@sanoursa, what are your thoughts on that approach?